src/main/java/com/farion/onlinebookstore/repository/book/IsbnSpecificationProvider.java

```diff
@@ -0,0 +1,19 @@
+package com.farion.onlinebookstore.repository.book;
+
+import com.farion.onlinebookstore.entity.Book;
+import com.farion.onlinebookstore.repository.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class IsbnSpecificationProvider implements SpecificationProvider<Book> {
+    @Override
+    public String getKey() {
+        return "isbn";
+    }
+
+    public Specification<Book> getSpecification(String param) {
+        return (root, query, criteriaBuilder) -> criteriaBuilder.like(
+                root.get("isbn"), "%" + param + "%");
+    }
```

Comment:
Actually strings 'title' and 'author' and 'isbn' is used in several classes, so I suggest creating class where you will keep constants and reuse them when it is required